### ISSUE TITLE
Refactor JmsPollableMessageChannelSpec for target channel impl generic type

### ISSUE THREAD

**Issue by artembilan** [spring-projects/spring-integration#3194]
See TODO: https://github.com/spring-projects/spring-integration/blob/master/spring-integration-jms/src/test/java/org/springframework/integration/jms/dsl/JmsTests.java#L374

Do the same with `AmqpPollableMessageChannelSpec`.

**Comment by oxcafedead** [comment]
Hi @artembilan, I've just looked in the code and was a bit confused by Specs hierarchy.
Maybe you could clarify some moments, I would appreciate it:
- I am not sure why pubsub channel extends pollable one. If it is just for code reuse, maybe it is better to use aggregation approach.
- Why do we need `JmsMessageChannelSpec` as an intermediate class between them? Initially I thought it was some generic class and both pubsub and pollable extend it but found that it's not really.

**Comment by artembilan** [comment]
Well, the problem is that all those channel implementations are really inherit from each other.
All of them rely on the `JmsChannelFactoryBean` and designing Spec for those channel impls, I realized that `SubscribableJmsChannel` is different from the `PollableJmsChannel` only with its own `container` option.
The Pub/Sub is just a variant of the `SubscribableJmsChannel`, but we decided to extract a separate Spec for it to give more fluent API to configure Pub/Sub options.

We may consider further refactoring over there as you describe, but probably we should do that in the next `6.0` version to provide less breaking right now...
